### PR TITLE
PPM Estimator: Use Google Maps Directions API to calculate distances

### PR DESCRIPTION
**Pull request by jamesatheyDDS**
## Checklist

I have…

- [x] run the application locally (`bin/rails server`) and verified that my changes behave as expected.
- [x] run static code analysis (`bin/rubocop`) and vulnerability scan (`bin/brakeman`) against my changes.
- [x] run the test suite (`bin/rake spec`) and verified that all tests pass.
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] read, understand, and agree to the terms described in [CONTRIBUTING.md](https://github.com/deptofdefense/move.mil/blob/master/CONTRIBUTING.md).

## Summary of Changes

This pull request…

- Uses the Google Maps Directions API to calculate distances between full ZIP codes in the PPM Estimator
- Adds the google-maps gem
- Removes the cached ZIP3-to-ZIP3 distances retrieved from DTOD from the database and seeds

It turns out that PPMs do not use Basepoint City to Basepoint City distance, like HHGs. Instead, DPS asks DTOD for the distance between the full ZIP5 codes to calculate the distance.

Since we don't want to rely on DTOD for the move.mil portal, this PR asks Google for the distance instead. It uses the same GOOGLE_MAPS_API_KEY environment variable as Locator Maps.

This is more accurate than the previously used ZIP3-to-ZIP3 distances, but is not identical to what DTOD would return for the same query, since DTOD uses a different data source and takes trucking restrictions into account.

Removing the stored ZIP3-to-ZIP3 distance data also reduces the time needed to seed.

One downside to this change is that estimates take slightly longer to calculate, due to the call to Google's API.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. git checkout `ppm-zip5-distance`,
1. set up development dependencies according to [CONTRIBUTING.md](https://github.com/deptofdefense/move.mil/blob/master/CONTRIBUTING.md),
1. run `bin/rails server`, and
1. load up [http://localhost:3000](http://localhost:3000) in the Web browser of your choice.

### CODE REVIEW

File: app/models/ppm_estimator.rb [hunk 1]
```diff
@@ -115,7 +115,7 @@ def dest_svc_area
   end
 
   def distance
-    @distance ||= DtodZip3Distance.dist_mi(start_zip3.zip3, end_zip3.zip3).to_i
+    @distance ||= (Google::Maps.route(estimator_params[:start], estimator_params[:end], key: Google::Maps.api_key).distance.value / 1609.344).round
```
Comment (jgarber623-gov) [inline app/models/ppm_estimator.rb]: What is `1609.344` ?

File: db/migrate/20180330171014_drop_dtod_zip3_distances.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class DropDtodZip3Distances < ActiveRecord::Migration[5.1]
+  def change
+    drop_table :dtod_zip3_distances
```
Comment (jgarber623-gov) [inline db/migrate/20180330171014_drop_dtod_zip3_distances.rb]: :clap:

File: config/initializers/google_maps_config.rb [hunk 3]
```diff
@@ -0,0 +1 @@
+Google::Maps.api_key = ENV['GOOGLE_MAPS_API_KEY'] || nil
```
Comment (jgarber623-gov) [inline config/initializers/google_maps_config.rb]: `ENV['GOOGLE_MAPS_API_KEY']` _should_ return `nil` if it isn't set, so you might not need the `|| nil` here.

File: app/models/ppm_estimator.rb (hunk 1 above)
Comment (jamesatheyDDS) [reply app/models/ppm_estimator.rb]: 1609.344 meters in a mile

**Pull request merged**